### PR TITLE
Change from IWorkState to CancellationToken.

### PR DESCRIPTION
**Pull request by ddunkin**
`IWorkState` will be removed by Faithlife/FaithlifeUtility#7.

### CODE REVIEW

File: src/Faithlife.WebRequests/WebServiceRequestBase.cs [hunk 1]
```diff
@@ -289,19 +264,19 @@ private HttpContent GetRequestContent(HttpRequestMessage webRequest)
...
 		{
 			// handle web response
-			WebServiceResponseHandlerInfo<TResponse> info = new WebServiceResponseHandlerInfo<TResponse>(webResponse, workState);
+			WebServiceResponseHandlerInfo<TResponse> info = new WebServiceResponseHandlerInfo<TResponse>(webResponse, cancellationToken);
 			try
 			{
 				if (!await HandleResponseCoreAsync(info).ConfigureAwait(false))
 				{
-					if (workState.Canceled)
+					if (cancellationToken.IsCancellationRequested)
```
Comment (bgrainger) [inline src/Faithlife.WebRequests/WebServiceRequestBase.cs]: I hate exceptions for flow control as much as (probably more than) the next guy, but should we be doing `cancellationToken.ThrowIfCancellationRequested()` here?

**Comment by ddunkin** [comment]
Is the synchronous `GetResult()` even appropriate? I'm inclined to remove it and let callers be explicit about going from async to sync.

File: src/Faithlife.WebRequests/WebServiceRequestBase.cs (hunk 1 above)
Comment (ddunkin) [reply src/Faithlife.WebRequests/WebServiceRequestBase.cs]: Agreed. Callers would expect cancellation to throw, not return null.

File: src/Faithlife.WebRequests/WebServiceRequestBase.cs [hunk 2]
```diff
@@ -145,52 +145,27 @@ protected WebServiceRequestBase(Uri uri)
...
-		}
-
-		/// <summary>
-		/// Gets the response.
-		/// </summary>
-		/// <param name="workState">The work state.</param>
+		/// <param name="cancellationToken">The token to monitor for cancellation requests.</param>
 		/// <returns>Makes a web request, blocking until the request has returned a response. Returns default(TResponse) if cancelled.</returns>
-		public TResponse GetResponse(IWorkState workState)
+		public TResponse GetResponse(CancellationToken cancellationToken = default(CancellationToken))
```
Comment (ejball) [inline src/Faithlife.WebRequests/WebServiceRequestBase.cs]: Supporting cancellation on sync methods is highly unusual. If we must keep the sync method, we should drop the cancellation token; clients that need both that block on the async method themselves.

File: src/Faithlife.WebRequests/Json/JsonWebServiceContent.cs [hunk 3]
```diff
@@ -102,7 +102,7 @@ protected virtual string GenerateJson()
 			// don't let StreamWriter close the stream
 			using (WrappingStream wrappingStream = new WrappingStream(stream, Ownership.None))
 			using (StreamWriter writer = new StreamWriter(wrappingStream))
-				await writer.WriteBatchedAsync(Json, 4096, WorkState.None).ConfigureAwait(false);
+				await writer.WriteAsync(Json).ConfigureAwait(false);
```
Comment (ejball) [inline src/Faithlife.WebRequests/Json/JsonWebServiceContent.cs:105]: https://github.com/dotnet/corefx/issues/9071

**Review by ejball — APPROVED** [review]

**Pull request merged**